### PR TITLE
[analyzer] Use diagtool to list clang warnings

### LINKED ISSUE #758: add option to list clang warning as checkers and allow them to be switched on/off
- [x] Enabling and disabling is available since v6.8.0 #1654
- [ ] listing will be done with `diagtool` see #1693

### LINKED ISSUE #1693: use diagtool to list clang warnings
The warnings supported by clang can not be listed with clang now, but there is a tool [`diagtool`](https://github.com/llvm-mirror/clang/blob/master/docs/CommandGuide/diagtool.rst) which could be used.

It is not part of the official installation yet, but it will be available from [clang 7](https://github.com/llvm-mirror/clang/blob/release_70/docs/ReleaseNotes.rst).

We should detect it the same way as we detect the clang installation (available in the PATH, absolute path in the config file, package relative path in the config file)

### PR DESCRIPTION
**Pull request by csordasmarton**
> Closes #758 and Closes #1693

### CODE REVIEW

**Comment by Xazax-hun** [comment]
Unfortunately, some packages do not contain diagtool, like https://packages.debian.org/sid/amd64/clang-7/filelist

As I can see this is not a problem since we only add the flag if diagtool is available. One question is though, how do we make sure we pick the right diagtool if multiple clang instances are available? We have somewhere some methods to pick the right clang. Shouln'd we do something similar for diagtool?

**Comment by gyorb** [comment]
The [clang-8 debian package](https://packages.debian.org/sid/amd64/devel/clang-8/filelist) still does not contain the diagtool binary.
I think the best option would be if there is an absolute or relative path in the configuration we should use that binary (the same as for clang and clang-tidy), however the user needs to set the right values.

If diagtool is searched from the PATH we should search and select clang first (which is done now) and check for a diagtool next to the selected clang binary. So if the diagtool will be packed next to a clang release it will be found there.

File: analyzer/codechecker_analyzer/cmd/checkers.py [hunk 1]
```diff
@@ -24,6 +27,64 @@
...
+    """
+    Return full path of diagtool.
+
+    Select clang binary, check for a 'diagtool' binary next to the selected
+    clang binary and return full path of this binary if it exists.
+    """
+    context = analyzer_context.get_context()
+    analyzer_binary = context.analyzer_binaries.get(ClangSA.ANALYZER_NAME)
+
+    clang_bin = find_executable(analyzer_binary,
```
Comment (gyorb) [inline analyzer/codechecker_analyzer/cmd/checkers.py]: It is possible that the configured clang binary is not in the PATH?
For example I've installed clang6 from the ubuntu package which is in the PATH, but I've built clang8 from source which I set in the config file. In that case will this find the right diagtool next to the clang8 build?

File: analyzer/codechecker_analyzer/cmd/checkers.py [hunk 2]
```diff
@@ -24,6 +27,64 @@
...
+              clang_bin)
+
+    return None
+
+
+def get_warnings(env=None):
+    """
+    Returns list of warning flags by using diagtool.
+    """
+    diagtool_bin = get_diagtool_bin(env)
```
Comment (gyorb) [inline analyzer/codechecker_analyzer/cmd/checkers.py]: Shouldn't we check if `None` is returned by `get_diagtool_bin`?

**Pull request merged**